lib/httplib/httpheaders.go

```diff
@@ -31,33 +31,25 @@ import (
 	"github.com/gravitational/teleport/api/utils"
 )
 
-// Mutex protected cache for memoizing functions which construct the CSP header string.
+// Cache for memoizing functions which construct the CSP header string.
 // This is necessary because the CSP header is constructed on every request to the web UI,
 // so caching here has a significant performance impact.
-type cspCache struct {
-	sync.RWMutex
-	entries map[string]string
-}
-
-func (c *cspCache) get(key string) (string, bool) {
-	c.RLock()
-	defer c.RUnlock()
-
-	val, ok := c.entries[key]
-	return val, ok
-}
-
-func (c *cspCache) set(key, value string) {
-	c.Lock()
-	defer c.Unlock()
+type cspCache struct{ sync.Map }
+
+func (c *cspCache) load(key string) (string, bool) {
+	if cspString, ok := c.Map.Load(key); ok {
+		if cspStringStr, ok := cspString.(string); ok {
+			return cspStringStr, true
+		} else {
+			panic(fmt.Sprintf("cspString is not a string: %v", cspString))
+		}
```

Comment:
You can omit the `, ok` and the else - the behavior will be exactly the same (panic)